### PR TITLE
Add support for "FOR UPDATE" and "SKIP LOCKED"

### PR DESCRIPTION
**Pull request by btubbs**
This addresses #43.  It's a first stab, but hopefully close.

The public interface only exposes a .ForUpdate() method on the dataset for now, but under the hood it's built to support .ForNoKeyUpdate(), .ForShare(), and .ForKeyShare() as well.

### CODE REVIEW

**Comment by btubbs** [comment]
Oh and this doesn't yet support the "OF table_name" optional bit of a FOR clause.

File: default_adapter.go [hunk 1]
```diff
@@ -10,55 +10,61 @@ import (
...
+	default_returning_fragment         = []byte(" RETURNING ")
+	default_from_fragment              = []byte(" FROM")
+	default_where_fragment             = []byte(" WHERE ")
+	default_group_by_fragment          = []byte(" GROUP BY ")
+	default_having_fragment            = []byte(" HAVING ")
+	default_order_by_fragment          = []byte(" ORDER BY ")
+	default_limit_fragment             = []byte(" LIMIT ")
+	default_offset_fragment            = []byte(" OFFSET ")
+	default_for_update_fragment        = []byte(" FOR UPDATE ")
+	default_for_no_key_update_fragment = []byte(" FOR NO KEY UPDATE ")
```
Comment (so1ar) [inline default_adapter.go]: This doesn't seem universally supported.  Does a downstream user just override this if their rdbms uses different syntax?

Comment (btubbs) [reply default_adapter.go]: Also, it's an opt-in feature.  If you write a query using it and your DB doesn't support it, you'll just get an error when trying to run the query.  You'll then just have to rewrite your query to not use the feature that your DB doesn't support.

**Comment by doug-martin** [comment]
@btubbs this looks good if you want to go ahead and add tests for the base adapter and the implemented adapters I can get this merged and released

**Comment by btubbs** [comment]
Cool.  I'll try to get that out this week.

**Comment by btubbs** [comment]
I added the other methods, and tests.

**Comment by arbrix** [comment]
@btubbs could you please tag your branch, this is allowed to use it at `replace` at `go mod` while this changes will be merged with the master.

**Comment by btubbs** [comment]
@arbrix I added the "for-update" tag.  Will that work?

**Pull request merged**